internal/controller/services/monitoring/monitoring_controller_actions.go

```diff
@@ -52,24 +51,27 @@ func initialize(_ context.Context, rr *odhtypes.ReconciliationRequest) error {
 // only when DSC has component as Managed and component CR is in "Ready" state, we add rules to Prom Rules.
 // all other cases, we do not change Prom rules for component.
 func updatePrometheusConfigMap(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
-	// Map component names to their rule prefixes
-	dsc, err := cluster.GetDSC(ctx, rr.Client)
-	// If the DSC doesn't exist, we don't need to update the prometheus configmap
-	if err != nil && k8serr.IsNotFound(err) {
-		return nil
-	}
+	_, err := cluster.GetDSCI(ctx, rr.Client)
```

Comment:
as far as I remember, the check on DSCI is performed by the reconciler, so this case should not happen

Reply:
you are right.
both calls of `GetDSCI()` can be removed here, already done in `Reconcile()`